CONTRIBUTING.md

```diff
@@ -1,60 +1,81 @@
+5. **Resolve Feedback**: Be open to feedback and iterate on your changes.
 
-```bash
-ssh-keygen -t ed25519 -C "<email-registered-on-github-account>"
-# upload public key to github
+### Quality Gates
 
-sudo dnf makecache --refresh
-sudo dnf install -y git-all
-sudo dnf install -y golang
-sudo dnf install -y podman
-sudo dnf install -y cri-o kubernetes-kubeadm kubernetes-node kubernetes-client cri-tools
-sudo dnf install -y operator-sdk
-sudo dnf install -y wget
-wget https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/linux/oc.tar.gz
-cd bin/; tar -xzvf ../oc.tar.gz ; cd .. ; rm oc.tar.gz
-sudo dnf install -y zsh
+To ensure the contributed code adheres to the project goals, we have set up some automated quality gates:
 
-# update PATH
-echo 'export PATH=${PATH}:~/bin' >> ~/.zshrc
-echo 'export GOPROXY=https://proxy.golang.org' >> ~/.zshrc
-```
+1. [linters](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/.github/workflows/linter.yaml): Ensure the check for linters is successful. If it fails, run `make lint` to resolve errors
+2. [api-docs](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/.github/workflows/check-file-updates.yaml): Ensure the api-docs are updated when making changes to operator apis. If it fails, run `make generate manifests api-docs` to resolve errors
+3. [unit-tests](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/.github/workflows/unit-tests.yaml): Ensure unit tests pass. Run `make unit-tests` 
+4. [e2e-tests](https://prow.ci.openshift.org/job-history/gs/test-platform-results/pr-logs/directory/pull-ci-opendatahub-io-opendatahub-operator-incubation-opendatahub-operator-e2e): Ensure CI job for [e2e tests](https://github.com/opendatahub-io/opendatahub-operator/tree/incubation/tests/e2e) pass. Refer run e2e locally to debug. CI test logs can also be found under `Artifacts` directory under Job details.
 
-## Using a local.mk file to override Makefile variables for your development environment
+### Code Style Guidelines
 
-To support the ability for a developer to customize the Makefile execution to support their development environment, you can create a `local.mk` file in the root of this repo to specify custom values that match your environment.
+1. Follow the Go community’s best practices, which can be found in the official [Effective Go](https://go.dev/doc/effective_go) guide.
+2. Follow the best practices defined by the [Operator SDK](https://sdk.operatorframework.io/docs/best-practices/).
+3. Use `go fmt` to automatically format your code.
+4. Ensure your code passes `make lint` (we have a .golangci.yml file configured in the repo).
+5. Ensure you write clear and concise comments, especially for exported functions.
+6. Always check and handle errors appropriately. Avoid ignoring errors by using _.
+7. Make sure to run `go mod tidy before` submitting a PR to ensure the `go.mod` and `go.sum` files are up-to-date.
```

Comment:
I also wonder -- do we have that `go mod tidy` in our PR checks (`git grep` does not find it but it might be executed some other way)? Or do we rely on folks being diligent with this?